library/src/scripts/components/forms/select/BigSearch.tsx

```diff
@@ -0,0 +1,156 @@
+import Option from "@library/components/forms/select/overwrites/Option";
+
+export interface IComboBoxOption {
+    value: string;
+    label: string;
+    data: any;
+}
+
+interface IProps {
+    query: string;
+    disabled?: boolean;
+    className?: string;
+    placeholder: string;
+    options?: any[];
+    loadOptions?: any[];
+    setQuery: (value) => void;
+}
+
+interface IState {
+    value: IComboBoxOption;
+}
+
+/**
+ * Implements the search bar component
+ */
+export default class BigSearch extends React.Component<IProps> {
+    public static defaultProps = {
+        disabled: false,
+    };
+
+    private id;
+    private prefix = "bigSearch";
+    private searchButtonID;
+    private searchInputID;
+
+    constructor(props) {
+        super(props);
+        this.id = getRequiredID(props, this.prefix);
+        this.searchButtonID = this.id + "-searchButton";
+        this.searchInputID = this.id + "-searchInput";
+    }
+
+    private handleOnChange = (newValue: any, actionMeta: any) => {
+        this.props.setQuery(newValue.label || "");
+    };
+
+    private handleInputChange = (newValue: any, actionMeta: any) => {
+        this.props.setQuery(newValue.label || "");
+    };
+
+    public render() {
+        const { className, disabled, options, loadOptions } = this.props;
+
+        /** The children to be rendered inside the indicator. */
+        const componentOverwrites = {
+            Control: this.BigSearchControl,
+            IndicatorSeparator: DoNotRender,
+            DropdownIndicator: DoNotRender,
+            ClearIndicator,
+            SelectContainer,
+            Menu,
+            MenuList,
+            Option,
+        };
+
+        const getTheme = theme => {
+            return {
+                ...theme,
+                borderRadius: {},
+                color: {},
+                spacing: {},
+            };
+        };
+
+        const customStyles = {
+            option: () => ({}),
+            menu: base => {
+                return { ...base, backgroundColor: null, boxShadow: null };
+            },
+        };
+
+        return (
+            <CreatableSelect
+                id={this.id}
+                inputId={this.searchInputID}
+                components={componentOverwrites}
+                isClearable={true}
+                isDisabled={disabled}
+                options={options}
+                classNamePrefix={this.prefix}
+                className={classNames(this.prefix, className)}
+                placeholder={this.props.placeholder}
+                aria-label={t("Search")}
+                escapeClearsValue={true}
+                pageSize={20}
+                theme={getTheme}
+                styles={customStyles}
+                backspaceRemovesValue={true}
+            />
+        );
+    }
+
+    public getValue = value => {
+        return value;
+    };
+
+    private BigSearchControl = props => {
```

Comment:
This should be it's own class component. Props deserves an annotation as well, here even if its any. The class comment for it should probably indicate that it for extending ReactSelect.

The primary reason to pull it into its own class component is because it has a callback method that is being defined on every render (`preventFormSubmission`) of this subcomponent.

The other reason is to make the stuff we are giving this ourselves (setQuery) and the its own props coming from react select distinct.